### PR TITLE
docs: Add some repl doc

### PR DESCRIPTION
**Pull request by dhylands**
This PR has 2 commits, the first moves the documentation for generating the docs into docs/README.md and the second adds some documentation for interrupting programs and paste mode.

### CODE REVIEW

**Comment by pfalcon** [comment]
Interrupting programs and paste mode are not specific to pyboard, but is generic feature. There rather should be port-independent "REPL reference" section.

**Comment by dhylands** [comment]
Yeah - I don't have my wipy yet so I wasn't sure how much was in common.

I agree it sounded generic and should probably go somewhere more common. I just wasn't sure exactly where.

**Comment by pfalcon** [comment]
It's common with even with unix and windows ports, so worth to be in a separate section (RST file), and just linked from Tutorial. As for under which ToC to include it is a question, currently "Language Reference" looks the best, though not ideal. But even just linking it from Tutorial will cover the original usecase, while paving path for more docs across ports.

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+Install Sphinx, and optionally (for the RTD-styling), sphinx_rtd_theme,
+preferably in a virtualenv:
+
+     pip install sphinx
+     pip install sphinx_rtd_theme
+
+In `micropython/docs`, build the docs:
+
+    make MICROPY_PORT=<port_name> BUILDDIR=<port_name>/build html
```
Comment (dpgeorge) [inline docs/README.md]: This was recently changed to build/<port_name> in 3833d3846db1e736a3b27455033dc7cc8939f828.

**Comment by dpgeorge** [comment]
I think "language reference" (docs/reference/index.html) is ideal for the REPL stuff.  The contents page says: Differences to standard Python as well as additional features of MicroPython are described in the sections here.

**Comment by dhylands** [comment]
I updated the path on the README.rst file

I created docs/reference/repl.rst and added a link from the reference/index.rst.

I also added a bunch more documentation.

File: docs/reference/repl.rst [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    >>> dir()
+    ['__name__', 'pyb']
+    >>>
+
+The special variable _ (underscore)
+-----------------------------------
+
+When you use the REPL, you may perfom computations and see the results.
+MicroPython stores the results of the previous statment in the variable _ (underscore).
+So you can use this to thn save that in a variable. For example:
```
Comment (dpgeorge) [inline docs/reference/repl.rst]: "So you can use the underscore to save the result in a variable.  For example:"

File: docs/reference/repl.rst [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+    paste mode; CTRL-C to cancel, CTRL-D to finish
+    === def foo():
+    ===     print('This is a test to show paste mode')
+    ===     print('Here is a second line')
+    === foo()
+    ===
+    This is a test to show paste mode
+    Here is a second line
+    >>>
+
```
Comment (dpgeorge) [inline docs/reference/repl.rst]: Maybe add a comment saying something like: "blank lines are allowed in pasted text, use ctrl-D to finish", and "the pasted text is compiled as if it were a file (this contrasts with single input statements for friendly REPL)".

File: docs/reference/repl.rst [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+    ...
+    Traceback (most recent call last):
+      File "<stdin>", line 3
+    IndentationError: unexpected indent
+
+If you press CTRL-E, then you will enter paste mode, which essentially turns off
+the auto-indent feature, and changes the prompt from ``>>>`` to ``===``. For example:
+
+    >>>
+    paste mode; CTRL-C to cancel, CTRL-D to finish
```
Comment (dpgeorge) [inline docs/reference/repl.rst]: In the latest version that should say "Ctrl-C" and "Ctrl-D".

File: README.md [hunk 5]
```diff
@@ -151,19 +151,9 @@ correct permissions.  Try then:
 
     $ sudo dfu-util -a 0 -d 0483:df11 -D build-PYBV10/firmware.dfu
 
-Building the documentation locally
-----------------------------------
+Documentation
```
Comment (dpgeorge) [inline README.md]: Do you think we need this small section at all?  If people want to compile the docs they'll probably anyway be looking in the docs dir and see the README.md there.

File: docs/reference/repl.rst (hunk 4 above)
Comment (dhylands) [inline docs/reference/repl.rst]: The repo still says CTRL:
https://github.com/micropython/micropython/blob/master/lib/utils/pyexec.c#L413
but I can change it....

Comment (dpgeorge) [inline docs/reference/repl.rst]: So it does!  Well I guess that needs changing to be consistent with unix port (but don't change the CTRL-B in raw repl parts, else pyboard.py stops working).

Comment (dpgeorge) [inline docs/reference/repl.rst]: Let me do it, it's easier.

**Comment by dhylands** [comment]
Summary of changes for this round:

- Removed Documentation note at the end of the main README.md
- Changed the title to "The MicroPython Interactive Interpreter Mode (aka REPL)"
and added a paragraph to explain REPL.
- Changed CTRL to Ctrl and also modified the code where it enters paste mode.
- Added a line about blank lines being added in paste mode.
- Used suggested wording for underscore example.

**Comment by dhylands** [comment]
And added some more verbage about paste mode (I reworded your suggestion slightly).

**Comment by dhylands** [comment]
rebasaed against the latest master

**Comment by dpgeorge** [comment]
I put the REPL link before the inline asm link in the references index, and merged.  Thank you!

**Pull request closed without merging**